modules/apps/headless-apio/person/person-apio-impl/src/main/java/com/liferay/person/apio/internal/architect/resource/base/BaseUserAccountCollectionResource.java

```diff
@@ -0,0 +1,247 @@
+			ThemeDisplay.class, CurrentUser.class
+		).addCreator(
+			this::addUser, ThemeDisplay.class, hasPermission::forAdding,
+			PersonCreatorForm::buildForm
+		).build();
+	}
+
+	@Override
+	public ItemRoutes<UserWrapper, Long> itemRoutes(
+		ItemRoutes.Builder<UserWrapper, Long> builder) {
+
+		return builder.addGetter(
+			this::_getUserWrapper, ThemeDisplay.class
+		).addRemover(
+			idempotent(userService::deleteUser), hasPermission::forDeleting
+		).addUpdater(
+			this::updateUser, ThemeDisplay.class, hasPermission::forUpdating,
+			PersonUpdaterForm::buildForm
+		).build();
+	}
+
+	@Override
+	public Representor<UserWrapper> representor(
+		Representor.Builder<UserWrapper, Long> builder) {
+
+		Representor.FirstStep<UserWrapper> builderFirstStep = builder.types(
+			"Liferay:UserAccount", "Person"
+		).identifier(
+			User::getUserId
+		).addDate(
+			"birthDate", BaseUserAccountCollectionResource::_getBirthday
+		).addLocalizedStringByLocale(
+			"honorificPrefix", _getContactField(Contact::getPrefixId)
+		).addLocalizedStringByLocale(
+			"honorificSuffix", _getContactField(Contact::getSuffixId)
+		).addNested(
+			"contactInformation", this::_getContact,
+			contactBuilder -> contactBuilder.types(
+				"ContactInformation"
+			).addString(
+				"facebookSn", ContactModel::getSmsSn
+			).addString(
+				"skypeSn", ContactModel::getSmsSn
+			).addString(
+				"twitterSn", ContactModel::getSmsSn
+			).build()
+		).addRelatedCollection(
+			"roles", RoleIdentifier.class
+		).addRelatedCollection(
+			"addresses", AddressIdentifier.class
+		).addRelatedCollection(
+			"emails", EmailIdentifier.class
+		).addRelatedCollection(
+			"phones", PhoneIdentifier.class
+		).addRelatedCollection(
+			"webUrls", WebUrlIdentifier.class
+		).addRelativeURL(
+			"image", UserWrapper::getPortraitURL
+		).addString(
+			"additionalName", User::getMiddleName
+		).addString(
+			"alternateName", User::getScreenName
+		).addString(
+			"dashboardURL", UserWrapper::getDashboardURL
+		).addString(
+			"email", User::getEmailAddress
+		).addString(
+			"familyName", User::getLastName
+		).addString(
+			"gender", BaseUserAccountCollectionResource::_getGender
+		).addString(
+			"givenName", User::getFirstName
+		).addString(
+			"jobTitle", User::getJobTitle
+		).addString(
+			"name", User::getFullName
+		).addString(
+			"profileURL", UserWrapper::getProfileURL
+		);
+
+		addCustomFields(builderFirstStep);
+
+		return builderFirstStep.build();
+	}
+
+	protected void addCustomFields(
+		Representor.FirstStep<UserWrapper> builderFirstStep) {
+	}
+
+	protected UserWrapper addUser(
+		PersonCreatorForm personCreatorForm, ThemeDisplay themeDisplay) {
+
+		return null;
+	}
+
+	protected abstract PageItems<UserWrapper> getPageItems(
+			Pagination pagination, FullNameQuery fullNameQuery,
+			Credentials credentials, ThemeDisplay themeDisplay,
+			CurrentUser currentUser)
+		throws PortalException;
+
+	protected UserWrapper updateUser(
```

Comment:
why?

Reply:
Can you return the builder in the base class and add the update/create part in the child classes?